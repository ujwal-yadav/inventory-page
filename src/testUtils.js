import { createTheme, ThemeProvider } from '@mui/material';
import { render } from '@testing-library/react';

export const renderWithTheme = (ui, { themeOptions = {} } = {}) => {
  const theme = createTheme(themeOptions);
  return render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>);
};
