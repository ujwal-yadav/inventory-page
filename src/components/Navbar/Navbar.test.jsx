import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { renderWithTheme } from '../../testUtils';
import { NavBar } from './Navbar';
import { fireEvent, screen } from '@testing-library/react';

describe('NavBar', () => {
  it('should render correctly', () => {
    const { asFragment } = renderWithTheme(<NavBar logOut={jest.fn()} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should call logOut when the button is clicked', () => {
    const mockLogOut = jest.fn();

    renderWithTheme(<NavBar logOut={mockLogOut} />);

    const button = screen.getByRole('button', { name: /log out/i });
    fireEvent.click(button);

    expect(mockLogOut).toHaveBeenCalledTimes(1);
  });
});
