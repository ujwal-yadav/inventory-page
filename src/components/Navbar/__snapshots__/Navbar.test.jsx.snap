// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should render correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-xd4mgo"
  >
    <div
      class="MuiBox-root css-axw7ok"
    >
      <div
        class="MuiBox-root css-34k09d"
      >
        <svg
          fill="none"
          height="100%"
          viewBox="0 0 40 40"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20 0c11.046 0 20 8.954 20 20v14a6 6 0 0 1-6 6H21v-8.774c0-2.002.122-4.076 1.172-5.78a10 10 0 0 1 6.904-4.627l.383-.062a.8.8 0 0 0 0-1.514l-.383-.062a10 10 0 0 1-8.257-8.257l-.062-.383a.8.8 0 0 0-1.514 0l-.062.383a9.999 9.999 0 0 1-4.627 6.904C12.85 18.878 10.776 19 8.774 19H.024C.547 8.419 9.29 0 20 0Z"
            fill="#007DFC"
          />
          <path
            d="M0 21h8.774c2.002 0 4.076.122 5.78 1.172a10.02 10.02 0 0 1 3.274 3.274C18.878 27.15 19 29.224 19 31.226V40H6a6 6 0 0 1-6-6V21ZM40 2a2 2 0 1 1-4 0 2 2 0 0 1 4 0Z"
            fill="#007DFC"
          />
        </svg>
      </div>
      <h1
        class="MuiTypography-root MuiTypography-h4 css-1n4lw5g-MuiTypography-root"
      >
        Bhumio
      </h1>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-u3uiol-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Log Out
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
